sgpt/role.py

```diff
@@ -113,22 +113,28 @@ def get_role_name(cls, initial_message: str) -> Optional[str]:
 
     @classmethod
     def _os_name(cls) -> str:
-        current_platform = platform.system()
-        if current_platform == "Linux":
-            return "Linux/" + distro_name(pretty=True)
-        if current_platform == "Windows":
-            return "Windows " + platform.release()
-        if current_platform == "Darwin":
-            return "Darwin/MacOS " + platform.mac_ver()[0]
-        return current_platform
+        if cfg.get("OVERWRITE_OS_NAME") != "":
+            return cfg.get("OVERWRITE_OS_NAME")
+        else:
+            current_platform = platform.system()
+            if current_platform == "Linux":
+                return "Linux/" + distro_name(pretty=True)
+            if current_platform == "Windows":
+                return "Windows " + platform.release()
+            if current_platform == "Darwin":
+                return "Darwin/MacOS " + platform.mac_ver()[0]
+            return current_platform
```

Comment:
```suggestion
        if cfg.get("OVERWRITE_OS_NAME") != "default":
            return cfg.get("OVERWRITE_OS_NAME")
        current_platform = platform.system()
        if current_platform == "Linux":
            return "Linux/" + distro_name(pretty=True)
        if current_platform == "Windows":
            return "Windows " + platform.release()
        if current_platform == "Darwin":
            return "Darwin/MacOS " + platform.mac_ver()[0]
        return current_platform
```

Reply:
I find calling "not overwriting" the "overwriting"-env var  "default" a bit confusing. I prefer the empty string or maybe something like "no_overwrite". This is personal taste though, if you insist on "default" of course, go ahead.

The else was indeed redundant, my bad.

Reply:
Let's have config variables as `OS_NAME` and `SHELL_NAME` with default values `auto`.